planemo/engine/test.py

```diff
@@ -1,23 +1,49 @@
+import os
+
 from planemo.engine import engine_context
-from planemo.galaxy import galaxy_config
+from planemo.galaxy import (
+    galaxy_config,
+    galaxy_serve,
+)
+from planemo.galaxy.api import DEFAULT_ADMIN_API_KEY
 from planemo.galaxy.config import _find_test_data
+from planemo.galaxy.ephemeris_sleep import sleep
 from planemo.galaxy.test import (
     handle_reports_and_summary,
     run_in_config,
 )
 from planemo.runnable import (
+    flatten_to_single_artifacts,
     for_paths,
     RunnableType,
 )
 
 
 def test_runnables(ctx, runnables, original_paths=None, **kwds):
     """Return exit code indicating test or failure."""
+    if kwds.get("serve"):
+        if "galaxy" not in kwds["engine"]:
+            raise ValueError("The serve option is only supported by Galaxy-based engines.")
+        kwds["galaxy_url"] = kwds["galaxy_url"] or "".join(("http://", kwds["host"], ":", str(kwds["port"])))
+        kwds["galaxy_admin_key"] = kwds["galaxy_admin_key"] or DEFAULT_ADMIN_API_KEY
+        pid = os.fork()
+        if pid == 0:
+            # wait for served Galaxy instance to start
+            sleep(kwds["galaxy_url"], verbose=ctx.verbose, timeout=500)
+            # then proceed to test against it
+            kwds["engine"] = "external_galaxy"
+        else:
+            # serve Galaxy instance
+            galaxy_serve(ctx, runnables, **kwds)
+            exit(1)
```

Comment:
Forking in the presence of threads seems brittle. With #1232 we're removing the run_tests.sh mode, so you could just sleep infinitely in cmd_test.sh / cmd_shed_test.sh after merging this.

Reply:
Thanks @mvdbeek, I'll have another look when #1232 gets merged.